odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -1435,3 +1435,22 @@ def test_move_N_cycle(self):
         """ Move multiple nodes to create a cycle. """
         with self.assertRaises(UserError):
             self.cats(1, 3).write({'parent': self.cats(9).id})
+
+
+class TestRequiredMany2one(common.TransactionCase):
+
+    def test_explicit_ondelete(self):
+        f = self.env['ir.model.fields']._get('test_new_api.req_m2o', 'foo')
```

Comment:
Why don't you check the field on the model instead?
```suggestion
        foo = self.env['test_new_api.req_m2o']._fields['foo']
```
This makes the test independent from the reflection of the field.